src/Microsoft.VisualStudio.Threading.Analyzers/CommonInterest.cs

```diff
@@ -43,5 +55,21 @@ public SyncBlockingMethod(IReadOnlyList<string> containingTypeNamespace, string
 
             public string AsyncAlternativeMethodName { get; private set; }
         }
+
+        internal struct LegacyThreadSwitchingMethod
+        {
+            public LegacyThreadSwitchingMethod(IReadOnlyList<string> containingTypeNamespace, string containingTypeName, string methodName)
```

Comment:
non-null arg validation in a struct has questionable value, since the default struct ctor is always present and leaves all properties in their null value states -- so having one public ctor with validation and readonly properties can give the mistaken impression that properties can't be null. So especially here where I can't depend on the Validation assembly, I tend to leaving it out.